filters/oscillatory_hallucination/filter.py

```diff
@@ -0,0 +1,243 @@
+from tasks.TaskTypes import TaskType
+
+
+class BasicTokenizer(object):
+    """This code is taken from the release of BERT, attributed to Devlin et al.
+    Runs basic tokenization (punctuation splitting, lower casing, etc.)."""
+
+    def __init__(
+        self,
+        do_lower_case=False,
+        never_split=None,
+        tokenize_chinese_chars=True,
+    ):
+        """Constructs a BasicTokenizer.
+        Args:
+            **do_lower_case**: Whether to lower case the input.
+            **never_split**: (`optional`) list of str
+                Kept for backward compatibility purposes.
+                Now implemented directly at the base class level (see :func:`PreTrainedTokenizer.tokenize`)
+                List of token not to split.
+            **tokenize_chinese_chars**: (`optional`) boolean (default True)
+                Whether to tokenize Chinese characters.
+                This should likely be deactivated for Japanese:
+                see: https://github.com/huggingface/pytorch-pretrained-BERT/issues/328
+        """
+        if never_split is None:
+            never_split = []
+        self.do_lower_case = do_lower_case
+        self.never_split = never_split
+        self.tokenize_chinese_chars = tokenize_chinese_chars
+
+    def whitespace_tokenize(self, text):
+        """Runs basic whitespace cleaning and splitting on a piece of text."""
+        text = text.strip()
+        if not text:
+            return []
+        tokens = text.split()
+        return tokens
+
+    def tokenize(self, text, never_split=None):
+        """Basic Tokenization of a piece of text.
+            Split on "white spaces" only, for sub-word tokenization, see WordPieceTokenizer.
+        Args:
+            **never_split**: (`optional`) list of str
+                Kept for backward compatibility purposes.
+                Now implemented directly at the base class level (see :func:`PreTrainedTokenizer.tokenize`)
+                List of token not to split.
+        """
+        never_split = self.never_split + (
+            never_split if never_split is not None else []
+        )
+        text = self._clean_text(text)
+        # This was added on November 1st, 2018 for the multilingual and Chinese
+        # models. This is also applied to the English models now, but it doesn't
+        # matter since the English models were not trained on any Chinese data
+        # and generally don't have any Chinese data in them (there are Chinese
+        # characters in the vocabulary because Wikipedia does have some Chinese
+        # words in the English Wikipedia.).
+        if self.tokenize_chinese_chars:
+            text = self._tokenize_chinese_chars(text)
+        orig_tokens = self.whitespace_tokenize(text)
+        split_tokens = []
+        for token in orig_tokens:
+            if self.do_lower_case and token not in never_split:
+                token = token.lower()
+                token = self._run_strip_accents(token)
```

Comment:
It looks like at least one of the differences from the BasicTokenizer in BERT is around these lines of code (Based on the BasicTokenizer class here:
https://huggingface.co/transformers/_modules/transformers/models/bert/tokenization_bert.html#BertTokenizer). 

It could be helpful to comment or document the changes that have been made so that we can easily understand why the BasicTokenizer class has to be copied out and used here instead of doing `from transformers import BasicTokenizer`.

Reply:
Thanks @timothy22000 for pointing this out. I was using the code from the original mbert release, upon closer look, just importing from transformers does the same task as well. I have updated my code with just the import.